### DISCUSSION TITLE
UCI  code  for "pass" in Janggi Game

### DISCUSSION THREAD

**Discussion by levis77** [fairy-stockfish/Fairy-Stockfish#636] [Q&A] [answered]
Hello Fabian, 

I am trying to use UCI  to run fairy stockfish, I need to set black side to "pass" the turn,   I tried both "0000","@@@@",  but it seems that engine still give suggestion for  black side, can you help me to figure out?
The engine I used is:-
NNUE evaluation using janggi-7fb17c3419dc.nnue enabled

1)position fen r2a1anbr/5k3/c6c1/n1Bp2p1p/9/9/1PP2PP1P/2N1C2C1/4K4/3A1A1NR w - - 0 10
go nodes 216
bestmove c7a10 ponder i10i9

2.1)position fen r2a1anbr/5k3/c6c1/n1Bp2p1p/9/9/1PP2PP1P/2N1C2C1/4K4/3A1A1NR w - - 0 10 moves c7a10 0000
go nodes 216
bestmove i7h7 ponder a10d8 ( i7h7 is for black side move)

2.2)position fen r2a1anbr/5k3/c6c1/n1Bp2p1p/9/9/1PP2PP1P/2N1C2C1/4K4/3A1A1NR w - - 0 10 moves c7a10
go nodes 216
bestmove i7h7 ponder e2e2  (Again, i7h7 is for black side move)

*Actually, I need white side move answer.

**Comment by ianfab** [comment, marked as answer]
Hi, passing moves in UCI are currently encoded with the origin and target squares of the king, e.g., `e2e2` in the starting position, similar as you would flip the king over the board to pass.

**Reply by levis77** [reply]
Thanks for you reply, it is working now.
